Design/WorkspacePropertyWithIndex.md

```diff
@@ -0,0 +1,137 @@
+
+- Maintaining consistency in how index types are presented and used.
+- Reduce the likelihood of errors due to incorrect use of `IndexInfo`.
+- Hide the details of MPI from the developer.
+
+## Current Situation
+The points are taken from the original `IndexInfo` design document linked in the first section.
+
+- There is no consistent way of defining index ranges or lists for algorithms.
+  This is done individually in each algorithm, and properties may be defined for spectrum numbers or workspace indices, and sometimes the interpretation of index properties even seems to depend on the facility (`LoadEventNexus`, see [13475](https://github.com/mantidproject/mantid/issues/13475)).
+- Validation of index ranges and behavior on errors such as out-of-range indices is likewise done individually in all algorithms.
+  - Algorithms behave inconsistently depending on errors (throw and error, print a warning, ignore silently).
+  - There are bugs in some range validations (see [15414](https://github.com/mantidproject/mantid/issues/15414)).
+  - Some algorithms do not validate for duplicate spectrum numbers and double-process the corresponding spectra (see [16651](https://github.com/mantidproject/mantid/issues/16651)).
+  - Validations are not unit-tested to a good extent.
+
+## Design Goals
+ - Provide a property type which can specify a workspace and allowed (and predefined) input index types.
+ - Unified behaviour on error (when bad indices are provided).
+ - Hide index translations from the developer (this will take place internally within the property).
+ - Automatically generate the required GUI input controls based on the specified index types.
+ - Return the workspace and correct index set simultaneously in one call to `PropertyManager::getProperty`. 
+ - Consistent and unit-tested translation without code duplication.
+ - Consistent and unit-tested validation without code duplication.
+ - Consistent and simple way of defining index properties without code duplication.
+ - Consistent user interface.
+ 
+## Implementation
+
+### Basics
+
+- Extend the `WorkspaceProperty` type.
+- Own an `ArrayProperty` which can make use of `ListPropertyWidget` for capturing list data from the GUI.
+- Add a set of flags which control allowed index types (for use in the constructor).
+- Use a `PropertyWithValue<std::string>` to capture the selected index type.
+
+### Details
+
+1. The `WorkspacePropertyWithIndex` type constructor will accept the following flags (or combinations of flags):
+	- `IndexType::SpectrumNumber` for list of global<sup>1</sup> spectrum numbers typically from 1-N.
+	- `IndexType::WorkspaceIndex` for list of global workspace indices typically from 0-N.
+	- `IndexType::DetectorID` for list of detector ids within the instrument.
+2. Users will be restricted to entering only one of the allowed index types at the GUI level. 
```

Comment:
I would think that they would forward the input to the algorithm property, after reading from something like a `QTextEdit`